mlserver/repository/repository.py

```diff
@@ -13,7 +14,17 @@
 DEFAULT_MODEL_SETTINGS_FILENAME = "model-settings.json"
 
 
-class ModelRepository:
+class ModelRepository(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    async def list(self) -> List[ModelSettings]:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    async def find(self, name: str) -> List[ModelSettings]:
+        raise NotImplementedError
+
+
+class ImplModelRepository(ModelRepository):
```

Comment:
Should we rename `ImplModelRepository` to something more descriptive about the repository implementation? The main feature of the current implementation is that it doesn't enforce any folder layout or schema, so I was thinking `SchemalessModelRepository`? Or maybe just `DefaultRepository`? Names are hard...

Reply:
I'll do it. SchemalessModelRepository

Reply:
Done